net/unbound/files/rootzone.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/sh
+##############################################################################
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# Copyright (C) 2016 Eric Luehrsen
+#
+##############################################################################
+#
+# This component needs to be used with the unbound.sh as an include. It uses
+# defaults and UCI scope variables define there. Function rootzone_update() is
+# intended to be part of a crontab or service stop to copy RFC5011 results
+# back to /etc/unbound/root.key.
+#
+##############################################################################
+
+rootzone_uci() {
+  # TODO: place holder; rootzone may be called out of context, so load few UCI
+  echo
+}
+
+##############################################################################
+
+roothints_update() {
+  # TODO: place holder; maybe we won't do this, but structure it in
+  echo
+}
+
+##############################################################################
+
+rootkey_update() {
+  if [ "$UNBOUND_N_ROOT_AGE" -lt 1 \
+      -o "$UNBOUND_N_ROOT_AGE" -gt 90 \
+      -o "$UNBOUND_B_DNSSEC" -lt 1 ] ; then
+    # Feature disabled
+    return 0
+  fi
+
+
+  state="$1"
+  drilltool=$( which drill )
+  digtool=$( which dig )
+  anchortool=$( which unbound-anchor )
```

Comment:
The path is known, so instead of shelling out to `which` you can do the following:

```
local anchortool
if [ -x /usr/bin/unbound-anchor ]; then
  anchortool=/usr/bin/unbound-anchor
else
  anchortool=
fi
```

She shell built-in `type` is usually sufficient even if you had to rely on a `PATH` lookup: `anchortool=$(type -p unbound-anchor || :)` (`:` is the so-called “null command”, `|| :` silently ignores errors by the first command; not an issue unless the shell is configured to exit immediately if a command exists with a non-zero status, i.e. `set -e`).